code/datums/components/food/bacteria_carrier.dm

```diff
@@ -0,0 +1,100 @@
+// Don't eat off the floor or hold parent object with dirty hands, you'll get sick
+
+/// Time needed for bacteria to infect the parent object
+#define FIVE_SECOND_RULE (5 SECONDS)
+/// Max number of symptoms on the random disease
+#define MAX_DISEASE_SYMPTOMS 2
+/// Max strength of the random disease
+#define MAX_DISEASE_STRENTH 3
+
+/datum/component/bacteria_carrier
+	/// Timer for counting delay before becoming infective
+	var/timer_id
+	/// Whether it is already infective
+	var/infective = FALSE
+
+/datum/component/bacteria_carrier/Initialize(mapload)
+	if(!isobj(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(examine))
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(handle_movement))
+	RegisterSignals(parent, list(
+		COMSIG_ITEM_DROPPED, //Dropped into the world
+		COMSIG_ATOM_EXITED), //Object exits a storage object (tables, boxes, etc)
+		PROC_REF(dropped))
+	RegisterSignals(parent, list(
+		COMSIG_ITEM_PICKUP, //Picked up by mob
+		COMSIG_ATOM_ENTERED), //Object enters a storage object (tables, boxes, etc.)
+		PROC_REF(picked_up))
+	handle_movement()
+
+/datum/component/bacteria_carrier/UnregisterFromParent()
+	UnregisterSignal(parent, list(
+		COMSIG_ATOM_EXAMINE,
+		COMSIG_MOVABLE_MOVED,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ATOM_EXITED,
+		COMSIG_ITEM_PICKUP,
+		COMSIG_ATOM_ENTERED,
+	))
+
+/datum/component/bacteria_carrier/Destroy()
+	remove_timer()
+	return ..()
+
+/datum/component/bacteria_carrier/proc/remove_timer()
+	if(!timer_id)
+		return
+	deltimer(timer_id)
+	timer_id = null
+
+/datum/component/bacteria_carrier/proc/handle_movement()
+	SIGNAL_HANDLER
+
+	if(infective)
+		return
+
+	var/obj/parent_object = parent
+	var/turf/open/open_turf = parent_object.loc
+
+	// Is parent on valid open turf?
+	if(!istype(open_turf) || islava(open_turf) || isasteroidturf(open_turf))
+		remove_timer()
+		return
+
+	// Is parent on an elevated structure?
+	for(var/atom/movable/content as anything in open_turf.contents)
+		if(GLOB.typecache_elevated_structures[content.type])
+			remove_timer()
+			return
```

Comment:
And pills should be considered as preserved food?

Reply:
So decomposition can be expanded to encompass both behaviors, perhaps via callbacks?

Or the "is on table" portion of decomposition can be separated out (perhaps into an element) that fire ssignals when an item is safe / unsafe?

There's just a bit of overhead to the current method of copy pasting decomposition behavior i'm not too plussed about

Reply:
>Or the "is on table" portion of decomposition can be separated out (perhaps into an element) that fire ssignals when an item is safe / unsafe?

Thanks for suggestion. I'll try to refactor it into a component like "germ_exposure" that will fire signals that are used by both decomposition and germ_carrier. It will probably lead to decomposition being affected by dirtyness of the tile or whether people walk on it, but this makes sense.